Source/Libraries/Adapters/GrafanaAdapters/Functions/BuiltIn/AddMetaData.cs

```diff
@@ -0,0 +1,77 @@
+﻿using GrafanaAdapters.DataSourceValueTypes;
+using GrafanaAdapters.DataSourceValueTypes.BuiltIn;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace GrafanaAdapters.Functions.BuiltIn;
+
+/// <summary>
+/// Returns a series with an extra MetaData Field.
+/// </summary>
+/// <remarks>
+/// Signature: <c>AddMetaData(field, value, expression)</c><br/>
+/// Returns: Series of values.<br/>
+/// Example: <c>AddMetaData('Company', 'GPA', FILTER ActiveMeasurements WHERE SignalType='FREQ')</c><br/>
+/// Execution: Deferred enumeration.
+/// </remarks>
+public abstract class AddMetaData<T> : GrafanaFunctionBase<T> where T : struct, IDataSourceValueType<T>
```

Comment:
To be consistent with other code naming / casing in the project, `AddMetaData` should be `AddMetadata`. When changing casing of name, be sure to change the name of the file also, which will likely require a `git  mv` operation in order for this to take effect.